File: 0704-binary-search/0704-binary-search.java
class Solution {
    public int search(int[] arr, int t) {
    int n = arr.length;
    int low=0,high=n-1;
    while(low<=high){
        int mid = (low+high)/2;
        if(arr[mid]==t){
            return mid;
        }
        else if (arr[mid]<t){
            low = mid+1;

        }
        else {
            high=mid-1;
        }
    }
    return -1;

    }
}

File: 0704-binary-search/0704-binary-search.cpp
class Solution {
public:
    int search(vector<int>& arr, int tar) {
        int low = 0  , high = arr.size()-1;
        
        while(low<=high){
            int mid=(low+high)/2;
            if(arr[mid]>tar){
                high = mid - 1;
            }
            else if(arr[mid]<tar){
                low = mid + 1;
            }
            else if(arr[mid]==tar){
                return mid;
            }

        }
        return -1;
    }
};